app/helpers/layout_helper.rb

```diff
@@ -122,7 +122,13 @@ def select_f(f, attr, array, id, method, select_options = {}, html_options = {})
     if include_blank
       blank_value = include_blank.is_a?(TrueClass) ? nil : include_blank
       blank_option = OpenStruct.new({id => '', method => blank_value })
+      frozen = false
```

Comment:
Good catch, but IMHO it's better to clone the array in any case, and not just if it's frozen. This way we will avoid modification of the source, which may lead to pretty unexpected side effects.
The problem is half-solved on the first line of the method, but for the case of arrays `to_a` returns `self`. We just need to treat it there.